crates/ssr/src/tests.rs

```diff
@@ -159,6 +159,50 @@ fn assert_match_failure_reason(pattern: &str, code: &str, snippet: &str, expecte
     assert_eq!(reasons, vec![expected_reason]);
 }
 
+#[test]
```

Comment:
All tests should pass at every commit, otherwise it can make bisection very hard.

I'd suggest either doing everything in one commit, or you could do the changes outside the ssr crate in one commit and the changes inside the srr crate in a second commit.

Reply:
Reordering the unit tests after all these commits should work for that too, right?